### ISSUE TITLE
missing some files and can not run gen_shadow

### ISSUE THREAD

**Issue by Dengshunge** [diffusion-face-relighting/difareli_code#4]
In `gen_shadow.py` file, missing file `shadow_cls_weight.pkl`, and can not run this scripts. 
Thank you.

**Comment by Dengshunge** [comment]
In `relight.py`, the ckpt model is None. However to specify?
![image](https://github.com/diffusion-face-relighting/difareli_code/assets/40339713/731cfcbb-731c-4344-b261-01ea1aaed37e)

**Comment by Dengshunge** [comment]
@XezXey

**Comment by XezXey** [comment]
I've add the shadow_cls_weight.pkl in this [commit](https://github.com/diffusion-face-relighting/difareli_code/commit/781c2c025100f925c689c1d072d9fb7fa64923b6).
For the ckpt model, you can specify the path to your model folder. Please refers to our documention in this [section](https://github.com/diffusion-face-relighting/difareli_code#checkpoints) for more details.

**Comment by anhdao69** [comment]
Hello @Dengshunge. Can you run diffae now?

**Comment by Dengshunge** [comment]
yes, the project can run successfully. @Johnx69

**Comment by anhdao69** [comment]
@Dengshunge  Thank you for replying me! 

Can you show me what is the ouput of gen_shadow? I got some values like this. Are they normal values?
 
![image](https://github.com/diffusion-face-relighting/difareli_code/assets/93436870/33bcc555-a857-419e-a710-b6e0dd91644a)

Also, one more thing about DECA. What is your CUDA version and g++ version to run? I run it on CUDA version 12.0 and it seems like the rasterization does not work. I got the black images after run the DECA module. Do you know how to fix this issue?

This is from file renderer.py

![Screenshot 2023-11-30 at 07 13 56](https://github.com/diffusion-face-relighting/difareli_code/assets/93436870/850b3f93-11f6-4c90-8579-052a47ef3f7b)

**Comment by Dengshunge** [comment]
@Johnx69 
1. the result of gen_shadow is right. I also get this valure.
2. I change the rasterizer of DECA. Same problem with you when using standard_rasterize. I recommend to use pytorch3d. you can change the code in `create_dataset.py` 
![image](https://github.com/diffusion-face-relighting/difareli_code/assets/40339713/b43475bb-570e-4d88-a067-f9fbd6f41a71)

**Comment by anhdao69** [comment]
Hello @Dengshunge, thank you very much! But I got another problem from the library. Can you give me your requirements file? and also the python version.

Thank you very much!

**Comment by Dengshunge** [comment]
@Johnx69 python version is 3.10.13
absl-py==2.0.0
aiohttp==3.8.6
aiosignal==1.3.1
albumentations==1.3.1
async-timeout==4.0.3
attrs==23.1.0
blobfile==2.1.0
cachetools==5.3.2
certifi==2023.7.22
charset-normalizer==3.3.1
chumpy==0.70
cmake==3.27.7
coloredlogs==15.0.1
contourpy==1.1.1
cycler==0.12.1
Cython==3.0.6
dlib==19.24.2
easydict==1.11
einops==0.7.0
face-alignment==1.4.1
filelock==3.12.4
flatbuffers==23.5.26
fonttools==4.43.1
frozenlist==1.4.0
fsspec==2023.10.0
future==0.18.3
fvcore==0.1.5.post20221221
google-auth==2.23.3
google-auth-oauthlib==1.1.0
grpcio==1.59.0
huggingface-hub==0.19.4
humanfriendly==10.0
idna==3.4
imageio==2.31.6
insightface==0.7
iopath==0.1.10
Jinja2==3.1.2
joblib==1.3.2
kiwisolver==1.4.5
kornia==0.7.0
lazy_loader==0.3
lightning-utilities==0.9.0
lit==17.0.3
llvmlite==0.41.1
lmdb==1.4.1
lpips==0.1.4
lxml==4.9.3
Markdown==3.5
MarkupSafe==2.1.3
matplotlib==3.8.0
mccabe==0.7.0
mpmath==1.3.0
multidict==6.0.4
networkx==3.2
ninja==1.11.1.1
numba==0.58.1
numpy==1.26.1
nvidia-cublas-cu11==11.10.3.66
nvidia-cuda-cupti-cu11==11.7.101
nvidia-cuda-nvrtc-cu11==11.7.99
nvidia-cuda-runtime-cu11==11.7.99
nvidia-cudnn-cu11==8.5.0.96
nvidia-cufft-cu11==10.9.0.58
nvidia-curand-cu11==10.2.10.91
nvidia-cusolver-cu11==11.4.0.1
nvidia-cusparse-cu11==11.7.4.91
nvidia-nccl-cu11==2.14.3
nvidia-nvtx-cu11==11.7.91
oauthlib==3.2.2
onnx==1.15.0
onnxruntime-gpu==1.16.3
opencv-python==4.8.1.78
opencv-python-headless==4.8.1.78
packaging==23.2
pandas==2.1.2
Pillow==10.0.1
portalocker==2.8.2
prettytable==3.9.0
protobuf==4.23.4
pyasn1==0.5.0
pyasn1-modules==0.3.0
pycodestyle==2.11.1
pycryptodomex==3.19.0
pyDeprecate==0.3.1
pyflakes==3.1.0
pyparsing==3.1.1
python-dateutil==2.8.2
pytorch-fid==0.3.0
pytorch-lightning==1.4.5
-e git+https://github.com/facebookresearch/pytorch3d.git@a8c70161a1c99c1878ae7cf312cf7907a84f01b0#egg=pytorch3d
pytz==2023.3.post1
PyYAML==6.0.1
qudida==0.0.4
requests==2.31.0
requests-oauthlib==1.3.1
rsa==4.9
safetensors==0.4.0
scikit-image==0.22.0
scikit-learn==1.3.2
scipy==1.11.3
six==1.16.0
sympy==1.12
tabulate==0.9.0
tensorboard==2.15.0
tensorboard-data-server==0.7.2
tensorboardX==2.6.2.2
termcolor==2.3.0
threadpoolctl==3.2.0
tifffile==2023.9.26
timm==0.9.11
torch==2.0.1
torchmetrics==0.5.0
torchvision==0.15.2
tqdm==4.66.1
triton==2.0.0
typing_extensions==4.8.0
tzdata==2023.3
urllib3==2.0.7
wcwidth==0.2.12
Werkzeug==3.0.1
yacs==0.1.8
yarl==1.9.2

**Comment by anhdao69** [comment]
@Dengshunge Thank you very much! Also, what is your CUDA version? I am still struggling with install pytorch3D

**Comment by Dengshunge** [comment]
@Johnx69 CUDA version is 12.0. 
I use the local installation method to install pytorch3d. 
`git clone https://github.com/facebookresearch/pytorch3d.git`
`cd pytorch3d && pip install -e .`

**Comment by anhdao69** [comment]
@Dengshunge where did you put the pytorch3d folder? Is this in `preprocess_scripts/Relighting_preprocessing_tools`?

**Comment by Dengshunge** [comment]
@Johnx69 It is unimportance. You can put it everywhere, because you use pip method to install pytorch3d.

**Comment by anhdao69** [comment]
I can run the code. But I still got black image rendered from DECA. Do you have this weight `preprocess_scripts/Relighting_preprocessing_tools/DECA/data/deca_model.tar` ? If yes, where did you download it?

**Comment by Dengshunge** [comment]
@Johnx69 file `deca_model.tar` is in the project of deca https://github.com/yfeng95/DECA. You can download release model. 
![image](https://github.com/diffusion-face-relighting/difareli_code/assets/40339713/bf4cec82-4c1c-42c5-a5c2-170435100bb8)

**Comment by anhdao69** [comment]
@Dengshunge, finally I got it. Thank you very much! I really appreciate your help!

**Comment by anhdao69** [comment]
Hello @Dengshunge, have you finished running the training code? Can you give me the requirements file and python version to run it? 

Thank you!

**Comment by Dengshunge** [comment]
I only run the inference code. @Johnx69

**Comment by anhdao69** [comment]
Thank you @Dengshunge. Did you successfully run the inference code? When I run, I saw that they require the dataset `ffhq_256_with_anno`. Do you know how to get that?

**Comment by xiangli1117** [comment]
@Dengshunge Hello, could you please tell me where you found this file? Or was it not used when generating shadow data? How did the code run successfully?

**Comment by xiangli1117** [comment]
Hello, could you please tell me where you found this file  ? Or was it not used when generating shadow data? How did the code run successfully?

**Comment by xiangli1117** [comment]
@Johnx69

**Comment by xiangli1117** [comment]
<img width="550" alt="屏幕截图 2024-08-20 182414" src="https://github.com/user-attachments/assets/85748a64-4355-447c-8ffc-87c9f83e2ea2">

**Comment by reremm2** [comment]
hello @xiangli1117,  im currently stuck in the same place where u have been. I really can't find out what the latent.pkl file is. Should i run another code do get it?

**Comment by xiangli1117** [comment]
Since it's been a while, I need to refresh my memory on the code before answering your question.



---Original---
From: ***@***.***&gt;
Date: Thu, Feb 20, 2025 15:37 PM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [diffusion-face-relighting/difareli_code] missing some files andcan not run gen_shadow (Issue #4)


   
hello @xiangli1117,  im currently stuck in the same place where u have been. I really can't find out what the latent.pkl file is. Should i run another code do get it?

—
Reply to this email directly, view it on GitHub, or unsubscribe.
You are receiving this because you were mentioned.Message ID: ***@***.***&gt;
 reremm2 left a comment (diffusion-face-relighting/difareli_code#4)
 
hello @xiangli1117,  im currently stuck in the same place where u have been. I really can't find out what the latent.pkl file is. Should i run another code do get it?

**Comment by reremm2** [comment]
wow that is really kind of you!!! I greatly appreciate your kindness

**Comment by xiangli1117** [comment]
The `latent.pkl` file is generated inside the `train` function. You can find the saving process in the source code of DiffAE at https://github.com/phizaz/diffae/blob/master/experiment.py.



叆叇
***@***.***



&nbsp;




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "diffusion-face-relighting/difareli_code"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2025年2月20日(星期四) 下午3:49
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [diffusion-face-relighting/difareli_code] missing some files and can not run gen_shadow (Issue #4)



   
wow that is really kind of you!!! I greatly appreciate your kindness

—
Reply to this email directly, view it on GitHub, or unsubscribe.
You are receiving this because you were mentioned.Message ID: ***@***.***&gt;
  reremm2 left a comment (diffusion-face-relighting/difareli_code#4)
 
wow that is really kind of you!!! I greatly appreciate your kindness

**Comment by reremm2** [comment]
think I figured it out!!! you saved me 😭 thanks!!!

**Comment by Thickdream** [comment]
> The `latent.pkl` file is generated inside the `train` function. You can find the saving process in the source code of DiffAE at https://github.com/phizaz/diffae/blob/master/experiment.py.
> 
> 
> 
> 叆叇
> ***@***.***
> 
> 
> 
> &nbsp;
> […](#)

Do you know how to run gen_shadow.py? It uses a dataset named “custom,” but this is not implemented within the framework.

**Comment by lllllialois** [comment]
> [@Dengshunge](https://github.com/Dengshunge) Thank you for replying me!
> 
> Can you show me what is the ouput of gen_shadow? I got some values like this. Are they normal values?
> 
> ![image](https://github.com/diffusion-face-relighting/difareli_code/assets/93436870/33bcc555-a857-419e-a710-b6e0dd91644a)
> 
> Also, one more thing about DECA. What is your CUDA version and g++ version to run? I run it on CUDA version 12.0 and it seems like the rasterization does not work. I got the black images after run the DECA module. Do you know how to fix this issue?
> 
> This is from file renderer.py
> 
> ![Screenshot 2023-11-30 at 07 13 56](https://github.com/diffusion-face-relighting/difareli_code/assets/93436870/850b3f93-11f6-4c90-8579-052a47ef3f7b)

@anhdao69  I'm running into the same issue where the DECA output results are just black images. Have you managed to find a solution for this yet?
